hspec-core/src/Test/Hspec/Core/Extension/Option.hs

```diff
@@ -0,0 +1,24 @@
+module Test.Hspec.Core.Extension.Option (
+  Option
+, flag
+, argument
+, noArgument
+, optionalArgument
+) where
+
+import qualified GetOpt.Declarative as Declarative
+import qualified Test.Hspec.Core.Config.Definition as Core
+
+import           Test.Hspec.Core.Extension.Config.Type (Option(..), Config)
+
+flag :: String -> (Bool -> Config -> Config) -> String -> Option
+flag name setter = Option . Core.flag name setter
+
+argument :: String -> String -> (String -> Config -> Maybe Config) -> String -> Option
```

Comment:
You are right about the which string is which. Perhaps an aliases would be helpful.
```haskell
argument :: CliName -> CliArgName -> CliHelp -> (Bool -> Config -> Config) -> SpecWith a
type CliName = String
type CliArgName = String
type CliHelp = String
```

I rearranged the arguments because I like having functions at the end which allows multiline lambdas without parenthaseses and hanging arguments. This should be applied to all the option functions if you choose to use it.

I changed `Option` to `SpecWith a` which removes the need for `registerOption[s]` and allows these to be placed as their own line in `do`. This is not a big deal but does seem to be cleaner and less overhead for extension writers. Applies to all option functions.